src/model/document.ts

```diff
@@ -268,14 +280,21 @@ export class Document implements Feature {
   }
 
   /**
-   * Adds and indexes a feature to this documentled before resolve().
+   * Return all warnings on local features contained within the document as well
+   * as warnings on features detected by following through all imports. If the
+   * deep argument is set to false, only return warnings found on local
+   * features.
    */
-  _addFeature(feature: Feature) {
-    if (this._doneResolving) {
-      throw new Error('_addFeature can not be called after _resolve()');
+  getWarnings(deep?: boolean): Warning[] {
+    const warnings: Warning[] = [];
+    if (deep == null) {
+      deep = true;
```

Comment:
I actually originally had that behavior as the default before realizing that all other methods (`getFeatures()`, `getByKind()`, `getById()`) either defaulted to using or only worked on the deep import graph. Having getWarnings() as the only method returning non-deep objects by default might trip users up who expect to be working with the entire import graph. wdyt?